frame/nfts/src/lib.rs

```diff
@@ -974,7 +1006,67 @@ pub mod pallet {
 			origin: OriginFor<T>,
 			collection: T::CollectionId,
 			item: T::ItemId,
-			maybe_check_delegate: Option<AccountIdLookupOf<T>>,
+			delegate: AccountIdLookupOf<T>,
+		) -> DispatchResult {
+			let maybe_check: Option<T::AccountId> = T::ForceOrigin::try_origin(origin)
+				.map(|_| None)
+				.or_else(|origin| ensure_signed(origin).map(Some).map_err(DispatchError::from))?;
+
+			let delegate = T::Lookup::lookup(delegate)?;
+
+			let collection_details =
+				Collection::<T, I>::get(&collection).ok_or(Error::<T, I>::UnknownCollection)?;
+			let mut details =
+				Item::<T, I>::get(&collection, &item).ok_or(Error::<T, I>::UnknownCollection)?;
+
+			let mut is_past_deadline = false;
+			let deadline = details.approvals.get(&delegate).ok_or(Error::<T, I>::NotDelegate)?;
+
+			if let Some(d) = deadline {
+				let now = frame_system::Pallet::<T>::block_number();
+				if *d < now {
+					is_past_deadline = true;
+				}
+			}
+
+			if !is_past_deadline {
```

Comment:
I think flipping the bool name is helpful here:
```suggestion
			let deadline = details.approvals.get(&delegate).ok_or(Error::<T, I>::NotDelegate)?;
            let is_before_deadline = if let Some(d) = deadline {
				let now = frame_system::Pallet::<T>::block_number();
				now <= *d
			} else { 
			    false 
			};

			if is_before_deadline {
```
(also we can set the bool in one go so don't need it to be mut.)

Reply:
Hmm..as for me, the code in the suggestion is less readable

Reply:
When you read/write the code you would probably think in that way: "Ok, if the deadline hasn't been passed, then we need to do some extra checks"

Reply:
That's cool if you prefer it that way. The other thought I had was maybe it would be nice to pull out the is passed deadline into a function as the same logic is in a couple of places?